graphics/managed_surface.h

```diff
@@ -30,143 +30,159 @@
 		const Common::Point &destPos, const uint32 *palette);
 
 	/**
-	 * Inner method for copying another surface into this one at a given destination position
+	 * Inner method for copying another surface into this one at a given destination position.
+	 * @todo there are a lot of parameters here, are they all fairly clear or should we document them?
 	 */
 	void transBlitFromInner(const Surface &src, const Common::Rect &srcRect,
 		const Common::Rect &destRect, uint transColor, bool flipped, uint overrideColor,
 		uint srcAlpha, const uint32 *srcPalette, const uint32 *dstPalette, const Surface *mask, bool maskOnly);
 public:
 	/**
-	 * Clips the given source bounds so the passed destBounds will be entirely on-screen
+	 * Clip the given source bounds so the passed destBounds will be entirely on-screen.
 	 */
 	bool clip(Common::Rect &srcBounds, Common::Rect &destBounds);
 public:
-	uint16 &w;
-	uint16 &h;
-	uint16 &pitch;
-	PixelFormat &format;
+	uint16 &w;           /*!< Width of the surface rectangle. */
+	uint16 &h;           /*!< Height of the surface rectangle. */
+	uint16 &pitch;       /*!< Pitch of the surface rectangle. See @ref Surface::pitch. */
+	PixelFormat &format; /*!< Pixel format of the surface. See @ref PixelFormat. */
 public:
 	/**
-	 * Create the managed surface
+	 * Create the managed surface.
 	 */
 	ManagedSurface();
 
 	/**
 	 * Create a managed surface from another one.
-	 * If the source surface is maintaining it's own surface data, then
-	 * this surface will create it's own surface of the same size and copy
-	 * the contents from the source surface
+	 *
+	 * If the source surface is maintaining its own surface data, then
+	 * this surface will create its own surface of the same size and copy
+	 * the contents from the source surface.
 	 */
 	ManagedSurface(const ManagedSurface &surf);
 
 	/**
-	 * Create the managed surface
+	 * Create the managed surface.
 	 */
 	ManagedSurface(int width, int height);
 
 	/**
-	 * Create the managed surface
+	 * Create the managed surface.
 	 */
 	ManagedSurface(int width, int height, const Graphics::PixelFormat &pixelFormat);
 
 	/**
-	 * Create the managed surface
+	 * Create the managed surface.
 	 */
 	ManagedSurface(ManagedSurface &surf, const Common::Rect &bounds);
 
 	/**
-	 * Destroy the managed surface
+	 * Destroy the managed surface.
 	 */
 	virtual ~ManagedSurface();
 
 	/**
-	 * Implements automatic conversion to a Graphics::Surface by
-	 * simply returning the inner surface. This must be const,
-	 * because we don't want changes being done directly to it,
-	 * since it would bypass dirty rect handling
+	 * Automatically convert to a Graphics::Surface by
+	 * simply returning the inner surface.
+	 *
+	 * This must be const, because changes are not supposed to be done
+	 * directly to it, since it would bypass dirty rect handling.
 	 */
 	operator const Surface &() const { return _innerSurface; }
+	/** @todo not sure what that rawSurface is */
```

Comment:
The `ManagedSurface` class is a wrapper around a `Surface` object. The `rawSurface` is simply this `Surface` object that is being wrapped.

Reply:
I copied the comment from the method above.

Question - is this technically an ``@overload`` if this method has a different name but does the same thing as the other method?

Reply:
Technically an overloaded functions are functions with the same name but different arguments. So no, it is not an overload.
Also, as you can see, the method above doesn't actually have a name. This is something called  a `conversion operator` in C++.